chain/ethereum/src/adapter.rs

```diff
@@ -143,6 +144,23 @@ impl bc::TriggerFilter<Chain> for TriggerFilter {
             traces: self.requires_traces(),
         }
     }
+
+    fn extend_with_template(
+        &mut self,
+        data_sources: impl Iterator<Item = <Chain as bc::Blockchain>::DataSourceTemplate>,
+    ) {
+        for data_source in data_sources.into_iter() {
+            self.log
+                .extend(EthereumLogFilter::from_mapping(&data_source.mapping));
+            self.call
+                .extend(EthereumCallFilter::from_mapping(&data_source.mapping));
+
+            let block = EthereumBlockFilter::from_mapping(&data_source.mapping);
+            if !block.is_empty() {
+                self.block.extend(block);
+            }
```

Comment:
Extending with an empty `block` is a noop. If the `if` is meant as an optimization, then it can be implemented within `extend`.